apisix/balancer/chash.lua

```diff
@@ -38,6 +38,9 @@ local function fetch_chash_hash_key(ctx, upstream)
         chash_key = ctx.var["http_" .. key]
     elseif hash_on == "cookie" then
         chash_key = ctx.var["cookie_" .. key]
+    elseif hash_on == "vars_combinations" then
+        chash_key = core.utils.resolve_var(key, ctx.var);
+        core.log.info("vars_combinations: " .. chash_key)
```

Comment:
`val, err = core.utils.resolve_var(key, ctx.var)`

this step may return `nil, err` case, we need to check the return value.

code style, here is a better one:

```lua
        chash_key = core.utils.resolve_var(key, ctx.var)
        core.log.info("vars_combinations: ", chash_key)
```

Reply:
and we need an error test case

Reply:
The error check is needed. But the test case can be omitted. Since there is no backtrack in the regex which used to match variable.

Reply:
@membphis @spacewander I will check the err , and just log the err message ,  is it ok?   I think I have nothing to do with it here,  refer to other codes using "resolve_var"

Reply:
Yes.